bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpContextFactoryService.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.net.http;
+
+import org.osgi.framework.Bundle;
+import org.osgi.service.http.HttpContext;
+import org.osgi.service.http.HttpService;
+
+/**
+ * Create {@link HttpContext} instances when registering servlets, resources or filters using the
+ * {@link HttpService#registerServlet} and corresponding methods.
+ *
+ * @author Henning Treu - initial contribution and API
+ *
+ */
+public interface HttpContextFactoryService {
+
+    /**
+     * Creates an {@link HttpContext} according to the OSGi specification of
+     * {@link HttpService#createDefaultHttpContext()} as described here:
+     * https://www.knopflerfish.org/releases/6.0.0/docs/javadoc/org/osgi/service/http/HttpService.html#createDefaultHttpContext
```

Comment:
We imho should not reference knopflerfish.org here - anyhow, the reference to the method of the interface should be good enough.